integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -1283,31 +1356,37 @@ fn test_resharding_v3_buffered_receipts_towards_splitted_shard_v2() {
 #[test]
 #[cfg_attr(not(feature = "test_features"), ignore)]
 fn test_resharding_v3_outgoing_receipts_towards_splitted_shard() {
+    let params = TestReshardingParameters::new();
     let receiver_account: AccountId = "account4".parse().unwrap();
     let account_1_in_stable_shard: AccountId = "account1".parse().unwrap();
     let account_2_in_stable_shard: AccountId = "account2".parse().unwrap();
+    let rpc_id = params.rpc_clients[0].clone();
     let params = TestReshardingParameters::new()
```

Comment:
Please avoid creating params twice, instead reuse the one that you have above. 

Some in other tests. 

nit: The following pattern may be a bit nicer:

```
let rpc_id = ...;
let params = TestReshardingParameters::new()
  .rpc_id(rpc_id)
  ...
```

Reply:
I wanted to avoid it, but `rpc_id` is calculated in `TestReshardingParameters::new()`.

Reply:
Mistakenly called `new()` twice in two tests, fixed it now.
But in general, I need:
```
    let params = TestReshardingParameters::new();
    let rpc_id = params.rpc_clients[0].clone();

    let params = params
        .add_loop_action(call_burn_gas_contract(
            ...
            rpc_id,
        ))
```
because `rpc_id` is calculated in `TestReshardingParameters::new()`.
Are you ok with that?

Reply:
@staffik I proposed to make `TestReshardingParameters` use a builder pattern to delay construction at the end instead of initializing everything in `new` (#12610). Maybe it can make applying your changes easier.

Reply:
That's amazing! Will wait for your PR to be merged in first.